com.unity.render-pipelines.universal/Runtime/Passes/DrawObjectsPass.cs

```diff
@@ -94,10 +94,31 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                 }
                 #endif
 
-                context.DrawRenderers(renderingData.cullResults, ref drawSettings, ref filterSettings, ref m_RenderStateBlock);
+                DrawingSettings drawSettings = CreateDrawingSettings(m_ShaderTagIdList, ref renderingData, sortFlags);
 
-                // Render objects that did not match any shader pass with error shader
-                RenderingUtils.RenderObjectsWithError(context, ref renderingData.cullResults, camera, filterSettings, SortingCriteria.None);
+                if((DebugHandler != null) && DebugHandler.IsActiveForCamera(ref renderingData.cameraData))
+                {
+                    foreach(DebugRenderSetup debugRenderSetup in DebugHandler.CreateDebugRenderSetupEnumerable(context, cmd))
+                    {
+                        DrawingSettings debugDrawingSettings = debugRenderSetup.CreateDrawingSettings(ref renderingData, drawSettings);
+
+                        if(debugRenderSetup.GetRenderStateBlock(out RenderStateBlock renderStateBlock))
+                        {
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref filterSettings, ref renderStateBlock);
+                        }
+                        else
+                        {
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref filterSettings, ref m_RenderStateBlock);
+                        }
+                    }
+                }
```

Comment:
How users that implement custom passes hook up with debug rendering? Seems too much to expect users to write this to their custom passes? Maybe this could be encapsulated.

Reply:
Yes, I believe this is basically it (I did this for GBufferPass to make some debug views work with deferred). I agree it's a handful. From what I understand, this code does 3 things:
1. Calls a setup function that sets up global state (e.g. GL.wireframe), and ensures it's reset afterwards
2. Overrides DrawingSettings if applicable for the active debug mode
3. Overrides RenderStateBlock if applicable for the active debug mode

I'm optimistic this could be made more user-friendly, but there are details I don't yet understand (this came up already with GBufferPass where there are multiple RenderStateBlocks which I didn't quite get).

I can see if we can simply make this a function to call, but I think it's a bigger task to ensure this works well with custom passes, and I'd suggest it's done separately later.

Reply:
Refactoring task: https://jira.unity3d.com/browse/XPIPELINE-156, agreed to look at this in a follow-up PR.